### PR TITLE
Yuetal

### PR DESCRIPTION
**Pull request by mmpagani**
This PR adds the Yu et al. (2013) GMPE used for the calculation of seismic hazard for China (2015 version). It currently implements the version supporting Ms. In a following PR, we will also add a version supporting Mw (with the use of an appropriate conversion equation). Tests are green: https://ci.openquake.org/job/zdevel_oq-engine/2653/

### CODE REVIEW

File: openquake/hazardlib/gsim/yu_2013.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+def rbf(ra, coeff, mag):
+    """
+    Calculate ground motion
+    """
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: The docstring should be extended with the parameter description:
```
:param ra: ...
:param coeff: ...
:param mag: ...
:returns: ...
```

File: openquake/hazardlib/gsim/yu_2013.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    term1 = a1ca + a1cb * mag + a1cc * np.log(ra + a1cd*np.exp(a1ce*mag))
+    term2 = a2ca + a2cb * mag
+    term3 = a2cd*np.exp(a2ce*mag)
+    return np.exp((term1 - term2) / a2cc) - term3
+
+
+def fnc(ra, *args):
+    """
+    Find the rupture to site distance
+    """
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: Same observation on the docstring

File: openquake/hazardlib/gsim/yu_2013.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                a1cb = C['b']
+                a1cc = C['c']
+                a1cd = C['d']
+                a1ce = C['e']
+                a2ca = C['ma']
+                a2cb = C['mb']
+                a2cc = C['mc']
+                a2cd = C['md']
+                a2ce = C['me']
+            #
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: Is this the same as the function define before? Could they be unified?

File: openquake/hazardlib/gsim/yu_2013.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            for epi, theta in zip(dists.repi, dists.azimuth):
+                res = minimize(fnc, epi, args=(epi, theta, mag, C),
+                               method='Nelder-Mead', tol=0.1)
+                ras.append(res.x[0])
+            ras = np.array(ras)
+            rbs = rbf(ras, C, mag)
+            #
+            # Compute values of ground motion for the two cases
+            mean1 = (a1ca + a1cb * mag + a1cc *
+                     np.log((ras**2+225)**0.5 + a1cd * np.exp(a1ce * mag)))
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: I think there is some PEP 8 violation here, not enough spaces around the binary operators.

File: openquake/hazardlib/gsim/yu_2013.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+            mean2 = (a2ca + a2cb * mag + a2cc *
+                     np.log((rbs**2+225)**0.5 + a2cd * np.exp(a2ce * mag)))
+            #
+            # Get distances
+            x = (mean1 * np.sin(np.radians(dists.azimuth)))**2
+            y = (mean2 * np.cos(np.radians(dists.azimuth)))**2
+            mean = mean1 * mean2 / np.sqrt(x+y)
+            if isinstance(imt, (PGA)):
+                mean = np.exp(mean)/g/100
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: Same PEP 8 violations here

File: openquake/hazardlib/gsim/yu_2013.py (hunk 3 above)
Comment (mmpagani) [reply openquake/hazardlib/gsim/yu_2013.py]: Done

File: openquake/hazardlib/gsim/yu_2013.py (hunk 4 above)
Comment (mmpagani) [reply openquake/hazardlib/gsim/yu_2013.py]: Thanks. Strangely this was not signalled by my editor.

File: openquake/hazardlib/gsim/yu_2013.py (hunk 5 above)
Comment (mmpagani) [reply openquake/hazardlib/gsim/yu_2013.py]: After the fix above I cannot find other PEP8 violations (also using this http://pep8online.com/)

**Comment by mmpagani** [comment]
We improved the docstring.

**Comment by mmpagani** [comment]
Hopefully, we addressed all the notes.

File: openquake/hazardlib/gsim/yu_2013.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    Returns the set of coefficients to be used for the calculation of GM
+    as a function of earthquake magnitude
+
+    :param coeff:
+        A dictionary of parameters for the selected IMT
+    :param mag:
+        Magnitude value
+    :return:
+        The set of coefficient to be used
+    """
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: coefficients

File: openquake/hazardlib/gsim/yu_2013.py [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+
+
+def rbf(ra, coeff, mag):
+    """
+    Calculate the meadian ground motion for a given magnitude and distance
+
+    :param ra:
+        Distance value [km]
+    :param coeff:
+        A dictionary of parameters for the
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: It seems the end of the sentence is missing

File: openquake/hazardlib/gsim/yu_2013.py [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+    term1 = a1ca + a1cb * mag + a1cc * np.log(ra + a1cd*np.exp(a1ce*mag))
+    term2 = a2ca + a2cb * mag
+    term3 = a2cd*np.exp(a2ce*mag)
+    return np.exp((term1 - term2) / a2cc) - term3
+
+
+def fnc(ra, *args):
+    """
+    Function used in the minimisation problem.
+
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: I think it is minimization

File: openquake/hazardlib/gsim/yu_2013.py [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    :param ra:
+        Semi-axis of the ellipses used in the Yu et al.
+    :return:
```
Comment (micheles) [inline openquake/hazardlib/gsim/yu_2013.py]: Wrong, it must be :returns:

Comment (mmpagani) [reply openquake/hazardlib/gsim/yu_2013.py]: Adjusted. I actually checked a few days ago online guidelines for docstrings; it is not clear (to me) if 'return' or 'returns' should be used.

**Pull request merged**